asf_core_data/pipeline/preprocessing/preprocess_epc_data.py

```diff
@@ -187,28 +195,61 @@ def load_and_preprocess_epc_data(
     # Do not save/overwrite the preprocessed data when not loading entire GB dataset
     # in order to prevent confusion.
 
+    if str(save_data) == "S3":
+        save_data = None
+    else:
+        raw_data_path = data_batches.get_batch_path(
+            base_config.RAW_EPC_DATA_PATH, data_path, batch, check_folder="input"
+        )
+
+        # Raw EPC can be found in data path?
+        raw_epc_exists = (Path(data_path) / raw_data_path).is_file()
+
     if n_samples is not None:
 
         save_data = None
         logging.warning(
             "You're not loading all samples so the processed data will not be saved!"
         )
 
-    if save_data or (save_data is not None and not os.path.isabs(save_data)):
+    if (save_data or (save_data is not None and not os.path.isabs(save_data))) and (
+        save_data is not None
+    ):
 
         save_data = data_path / save_data
 
         if subset != "GB":
             logging.warning("Careful! You're not loading the complete GB dataset.")
 
-    epc_df = epc_data.load_raw_epc_data(
-        data_path=data_path,
-        rel_data_path=rel_data_path,
-        subset=subset,
-        batch=batch,
-        usecols=usecols,
-        n_samples=n_samples,
-    )
+    if subset != "GB" and n_samples is not None:
+        logging.warning(
+            "Nation subsets do not work well in combination with low n_samples. Set n_samples=None for best results."
+        )
+
+    # Load raw data (from 'preprocessed' raw data or 'untouched' raw data)
+    # Shouldn't make a difference but sometimes useful when debugging
+    if (raw_epc_exists and not reload_raw) or str(data_path) == "S3":
+
+        # Load raw EPC from preprocessed batch (England/Wales/Scotland combines in EPC_GB_raw.csv)
+        epc_df = epc_data.load_preprocessed_epc_data(
+            data_path=data_path,
+            subset=subset,
+            batch=batch,
+            version="raw",
+            usecols=usecols + ["COUNTRY"],
+            n_samples=n_samples,
+        )
+
+    else:
+        # Load raw EPC data from England/Wales and Scotland data from raw data folders
+        epc_df = epc_data.load_raw_epc_data(
+            data_path=data_path,
+            rel_data_path=rel_data_path,
+            subset=subset,
+            batch=batch,
+            usecols=usecols,
+            n_samples=n_samples,
+        )
```

Comment:
I think there's a problem with indentation in this block (including the code below in lines 254 - 261 just before the return, that I can't select because they haven't been changed in this PR).
Might this be what it should look like (print below)? I also think there's still a part of the code that's missing, as highlighted with "SOMETHING STILL MISSING HERE". Sorry if I totally got this wrong.
<img width="708" alt="image" src="https://user-images.githubusercontent.com/70645959/223454359-95b52d0a-d451-4dfb-ac2f-2c2fd891b2b4.png">

Reply:
Ehrm, no, I think I got it right. 

- The subset check only triggers a warning, doesn't have any further consequences. So the if-block ends there.
- Next, we load the raw EPC data, either loading from inputs (reading an concatenating again all the EPC files) or dirctly from outputs where we have a merged version for all countries. 
- Then, we move that loaded raw epc data to the next function for processing.

No need for a nested if-else. Load the raw EPC data from one place or another, then pass to the processing funciton. As simple as that. Will add a few comments since this might be unclear.

Reply:
hey @athrado - sorry to unresolve this again, don't mean to be a pain, just want to make sure I understand this :)

I think I understand it now, but let me just ask a few more questions to make sure. I think two things got me confused initially:
- The comment in line 244: `# Load raw EPC data from England/Wales and Scotland data from raw data folders`. Maybe altering it to `# Load raw EPC data from England/Wales, Scotland or all GB data from raw data folders` would make it more obvious that it can be any of these subsets. Maybe also changing the comment `# Load raw EPC from preprocessed batch (England/Wales/Scotland combines in EPC_GB_raw.csv)` in line 233 could be of help.

- From what I understand now, the function `load_preprocessed_epc_data()` (used in line 234) doesn't really load the fully preprocessed epc data (the data still needs to be further preprocessed) - it only concatenates the raw files, is this right? That was a bit confusing because then you have the function call to `preprocess_data(). Would it be possible to rename the first function to something like `load_concatenated_raw_epc_data()` or something like this? I know you have a comment above referring to raw data, but it's still a bit confusing. Sorry if I got this wrong again😂

Another question I have is why do we have `usecols=usecols + ["COUNTRY"]` in line 239 but only `usecols=usecols` in line 250?

Reply:
That's what the 'unresolve' option is for! ;-) 

The thing with `load_preprocessed_epc_data()` is that it can be used to load the EPC data at any of the processing stages: raw, processed, processed and deduplicated. The raw data is indeed still unprocessed, but it has been loaded from the invidual original EPC files and been concatenated. The only other thing that was added was the 'COUNTRY' feature, which should answer your other question (good catch by the way): The raw data we load from the processed outputs dir already has a country tag (so we can/should load it), but the raw EPC data loaded from the unprocessed inputs dir doesn't yet have the COUNTRY feature. It will be added, though, as the EPC data is loaded one by one. Either way, you wind up with the EPC containing the feature COUNTRY.

So I see your concern about the name of `load_preprocessed_epc_data()` but since it's relatively rare that we use it to load the raw data (it's really just a speed boost in this case), I'd rather keep the name as it's quite helpful to get the processed EPC data in other cases. 

I've already altered some of the comments base on your last comment, but will have another look to make it more clear. If you still find it confusing, we could write a wrapper function called `load_concat_raw_epc_data()` which triggers a call to `load_preprocessed_epc_data('raw')`. Somewhat redunant, but could be an easy workaround if you feel the comments alone won't suffice.

Reply:
![Screenshot 2023-03-09 at 17 51 52](https://user-images.githubusercontent.com/42718928/224113177-326e0b33-427f-4285-9473-0524834c834d.png)

Just as a reminder how the 'processed' data is stored. You can see that raw is also put into that folder. We have had a discussion whether raw should go into inputs or outputs but decided to put it into outputs together with the others as it has been processed in some way.

Reply:
Okay, I understand it better now @athrado - thank you for explaining. Do keep the name of the function as is, it makes sense!
I agree about the location of the `raw` - because it's not really raw anymore... so it makes sense to be in outputs.